Assets/Plugins/Source/Editor/Utility/UnityProjectUtility.cs

```diff
@@ -122,51 +136,56 @@ private static bool CanFindMissingScriptsInScenes()
+            HashSet<string> checkedPrefabGuids = new();
 
-            HashSet<string> s_referencedPrefabGuids = new();
-            HashSet<string> s_checkedPrefabGuids = new();
+            // Get the path to the currently open scene so we can reopen it
+            // when we are done looking for unreferenced prefabs.
+            var previouslyOpenScene = SceneManager.GetActiveScene().path;
 
             // First pass: Find prefabs referenced in scenes
             foreach (var scenePath in EditorBuildSettingsScene.GetActiveSceneList(EditorBuildSettings.scenes))
             {
-                var scene = EditorSceneManager.OpenScene(scenePath, OpenSceneMode.Additive);
-                FindPrefabsInCurrentScene(ref s_referencedPrefabGuids);
+                var scene = EditorSceneManager.OpenScene(scenePath, OpenSceneMode.Single);
+                FindPrefabsInCurrentScene(ref referencedPrefabGuids);
             }
 
-            Debug.Log($"There are a total of {s_referencedPrefabGuids.Count} prefabs directly referenced by scenes.");
+            EditorSceneManager.OpenScene(previouslyOpenScene, OpenSceneMode.Single);
+
+            Debug.Log($"There are a total of {referencedPrefabGuids.Count} prefabs directly referenced by scenes.");
 
             // Second pass: Find prefabs referenced by other prefabs
             Debug.Log($"Now checking each prefab for prefab dependencies.");
-            var prefabsReferencedByScenesDirectory = new HashSet<string>(s_referencedPrefabGuids);
+            var prefabsReferencedByScenesDirectory = new HashSet<string>(referencedPrefabGuids);
             foreach (var prefabGuid in prefabsReferencedByScenesDirectory)
             {
-                FindPrefabDependencies(ref s_checkedPrefabGuids, ref s_referencedPrefabGuids, prefabGuid);
+                FindPrefabDependencies(ref checkedPrefabGuids, ref referencedPrefabGuids, prefabGuid);
             }
 
             // Compare to list of all prefabs
             string[] allPrefabGuids = AssetDatabase.FindAssets("t:Prefab");
             Debug.Log(
-                $"Comparing discovered (referenced) prefabs ({s_referencedPrefabGuids.Count}) to all prefabs ({allPrefabGuids.Length}).");
+                $"Comparing discovered (referenced) prefabs ({referencedPrefabGuids.Count}) to all prefabs ({allPrefabGuids.Length}).");
             foreach (var prefabGuid in allPrefabGuids)
             {
-                if (!s_referencedPrefabGuids.Contains(prefabGuid))
-                {
-                    string path = AssetDatabase.GUIDToAssetPath(prefabGuid);
-                    Debug.Log("Unreferenced Prefab: " + path);
-                }
+                // Skip if the prefab is referenced
+                if (referencedPrefabGuids.Contains(prefabGuid))
+                    continue;
+
+                string path = AssetDatabase.GUIDToAssetPath(prefabGuid);
+                Debug.LogWarning($"Unreferenced Prefab: {path}");
```

Comment:
thinking out loud, not essentially a thing that needs to be changed: This does exactly as it is supposed to, find all prefabs without references. But what about things that get loaded from Streaming Assets by their file path? Is there a world where this search is constrained to not search inside of Resources and Streaming Assets folders?

And what if the only reference to it is in Addressable bundles? E.x. I have an addressables bundle that contains GameObjects with cosmetic information in it. Those prefabs aren't referenced elsewhere in the game, just in being downloaded by their bundle...

Anyways that all said, this tool seems useful, and we already had it. Perhaps people who will utilize it will already be aware of the limitations of actually trying to answer "is this resource used by anything".

Reply:
This is an interesting point, I will keep it in mind.